### ISSUE TITLE
关于bs=1且多图输入时vision_tensors维度与count_vision_proj不匹配问题

### ISSUE THREAD

**Issue by CgggggY** [jingyaogong/minimind-v#114]
您好，我在阅读和调试 MiniMind-V 的 model_vlm.py 时，发现多图分支在单样本场景下可能存在一个维度组织不一致的问题。

问题位置大致在 forward() 中这段逻辑：
stack_dim = 1 if bs > 1 else 0
vision_tensors = torch.stack(
    [
        self.vision_proj(
            MiniMindVLM.get_image_embeddings(pixel_values[:, i, :, :, :], self.vision_encoder)
        )
        for i in range(num)
    ],
    dim=stack_dim
)
这里列表中每一项的 shape 是：[B, target_tokens, hidden_size]

因此：当 bs > 1 时，stack_dim = 1，得到的 vision_tensors 为：[B, N, target_tokens, hidden_size]
           当 bs = 1 时，stack_dim = 0，得到的 vision_tensors 为：[N, 1, target_tokens, hidden_size]

但是后续 count_vision_proj() 中对 vision_tensors 的访问方式是：
if k < vf.size(1):hb = torch.cat((hb[:start], vf[b][k][:i - start], hb[i:]), dim=0)[:seqlen]

这里明显假设 vf 的前两维语义是：[batch_size, num_images, ...]
也就是说，后续逻辑默认：vf.size(1) 表示该样本对应的图片数；vf[b][k] 表示第 b 个样本的第 k 张图片

因此：在 bs > 1 时，上下文是自洽的
           但在 bs = 1 且 num_images > 1 时，前面生成的是 [N, 1, ...]，此时 vf.size(1) = 1，不再表示图片数，后续 k < vf.size(1) 与 vf[b][k] 的语义都会出现错位，导致只能正确处理第一张图片

我这边的理解是，这里的 vision_tensors 应该统一保持为：[B, N, target_tokens, hidden_size]，这样才能与 count_vision_proj() 中的索引逻辑保持一致。一个比较直接的修复方式是：无论 bs 是否为 1，都统一在 dim=1 上进行 stack，即始终输出 [B, N, ...] 的布局。如果这个判断方向是对的，我这边已经做了对应修改，并可以提交 PR。

感谢开源！

**Comment by jingyaogong** [comment]
你说得对，可以提交PR

当前版本实际不支持多图，数据只有单图

**Comment by jingyaogong** [comment]
已合并，先close了😀

**Comment by CgggggY** [comment]
好的没问题！
